assets/js/accuracy-tool.js

```diff
@@ -0,0 +1,385 @@
+/* global mapboxgl */
+const accuracyApiUrl = 'https://web.solcast.com.au/api/accuracy';
+const accuracyVariant = 'historic';
+const accuracyToolContainer = document.getElementById('accuracyTool');
+const accuracyFilterForm = document.getElementById('accuracyFilter');
+const summaryContainer = document.getElementById('accuracySummary');
+const resultsContainer = document.getElementById('accuracyStatistics');
+const legendContainer = document.getElementById('accuracyMapLegend');
+const accuracyRequestDataUrl = accuracyToolContainer.getAttribute('accuracy-data-url');
+
+mapboxgl.accessToken = 'pk.eyJ1Ijoic29sY2FzdCIsImEiOiJjbHBrbmZybjUwMXhuMm5wZHZkYTl2cHgzIn0.JoR-him1ia9CPTHOTTNIXw';
+const map = new mapboxgl.Map({
+  container: 'accuracyMap',
+  maxZoom: 6,
+  minZoom: 1,
+  zoom: 1,
+  center: [40, 10],
+  bounds: [[-180, -75], [180, 75]],
+  style: 'mapbox://styles/solcast/clyzom3fx00ee01r5fw927na8',
+  attributionControl: false,
+});
+
+function renderMapLegend() {
+  legendContainer.innerHTML = '';
+  const legendItems = [
+    { label: 'Tropical', color: '#f26e54' },
+    { label: 'Desert', color: '#fcb82a' },
+    { label: 'Steppe', color: '#838bc2' },
+    { label: 'Temperate', color: '#98d09d' },
+    { label: 'Cold', color: '#60a9d1' },
+    { label: 'Polar', color: '#136aae' },
+  ];
+  legendItems.forEach((item) => {
+    const legendItem = document.createElement('div');
+    legendItem.className = 'map-legend-item';
+    const color = document.createElement('div');
+    color.className = 'map-legend-color';
+    color.style.backgroundColor = item.color;
+    legendItem.appendChild(color);
+    const label = document.createElement('div');
+    label.className = 'map-legend-label';
+    label.innerHTML = item.label;
+    legendItem.appendChild(label);
+    legendContainer.appendChild(legendItem);
+  });
+}
+
+function loadMapGeoJson(data) {
+  if (!map.getSource('sites')) {
+    map.addSource('sites', {
+      type: 'geojson',
+      data,
+    });
+  } else {
+    map.getSource('sites').setData(data);
+  }
+  if (!map.getLayer('sites')) {
+    map.addLayer({
+      id: 'sites',
```

Comment:
This object syntax is going to be rejected if we use the rules from https://github.com/Solcast/static/pull/8/files#r1713206672

We can merge yours first and then @gino-solcast 's - then consolidate the differences?